### PR TITLE
Implement a conditional request plugin interface so access requests can be automatically approved or denied

### PR DESCRIPTION
**Pull request by somethingnew2-0**
Includes a new plugin interface `ConditionalAccessPluginSpec` for allowing custom logic for conditionally automatically approving or denying access requests.

Breaking changes:
- `NotificationPluginSpec.access_request_created()` and `NotificationPluginSpec.access_request_completed()` were incorrectly sending the `OktaGroup.name` string as the `group` parameter instead of the `OktaGroup` object itself. This has been corrected so now the `OktaGroup` object is correctly sent to notification plugins for those functions.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207094387950525

### CODE REVIEW

**Comment by jonathanhle** [comment]
Just dropping a note that this is 🔥

File: examples/plugins/access_requests/requests.py
Comment (eguerrant) [inline examples/plugins/access_requests/requests.py:20]: Will the groups have to be called 'Auto-Approved-Group' if we want to allow auto approval..? Maybe I'm missing how this fits in with the rest of the code...

**Review by eguerrant — DISMISSED** [review]

File: examples/plugins/access_requests/requests.py
Comment (somethingnew2-0) [reply examples/plugins/access_requests/requests.py:20]: This is just a toy example of auto-approving membership to a single group. We can change the plugin code to approve groups by their name or some other criteria.

**Review by eguerrant — APPROVED** [review]

**Pull request merged**
